common/base.py

```diff
@@ -0,0 +1,77 @@
+from abc import ABC, abstractmethod
+
+import torch
+
+from common.memory_tracker import MemoryTracker
+
+
+class BaseBenchmarkClass(ABC):
+    def __init__(self, model_path: str, precision: int, device: str) -> None:
+        self.model_path = model_path
+        self.precision = precision
+        self.device = device
+
+        # tps = token/sec
+        self.tps_results = []
+        self.memory_usage_results = []
+        self.memory_tracker = MemoryTracker()
+
+    def preprocess_cuda(self, prompt: str):
+        input_tokens = self.tokenizer.encode(prompt, return_tensors="pt").to(
+            self.device
+        )
+        return {"input_tokens": input_tokens, "num_tokens": len(input_tokens[0])}
```

Comment:
this function assumes that tokenizer is initialized.  I would call initially a self.init_tokenizer() and create an abstract method self.init_tokenizer

Reply:
> this function assumes that tokenizer is initialized. I would call initially a self.init_tokenizer() and create an abstract method self.init_tokenizer

Yes, so I have a function called `load_model` where we load the model and tokenizer. I can rename the function to `load_model_and_tokenizer` Does that work?